### PR TITLE
Improve the release documentation.

### PR DESCRIPTION
**Pull request by nathanhammond**

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,16 +1,36 @@
...
+> I'm starting an Ember CLI release. Please hold off merging PRs, "homu r+"-ing, and pushing new code!
+
+### Environment
+
+Make sure that you're running the most recent stable `node` and bundled `npm`.
+
+```sh
+$ node --version
+$ npm --version
+$ cd ~/repos/ember-cli
```
Comment (kellyselden) [inline RELEASE.md]: I don't think this is terribly useful since it's a link to your filesystem.

File: RELEASE.md [hunk 2]
```diff
@@ -1,16 +1,36 @@
...
+$ git clean -fdx
```
Comment (kellyselden) [inline RELEASE.md]: this step is already at https://github.com/ember-cli/ember-cli/blob/master/dev/prepare-release#L3

File: RELEASE.md [hunk 3]
```diff
@@ -66,24 +96,40 @@ Test published version
...
+
+### Twitter
+
+> Ember CLI vX.Y.Z "Release name goes here." released!
+https://github.com/ember-cli/ember-cli/releases/tag/vX.Y.Z
+\#emberjs
+
+### Slack
+
+> Ember CLI vX.Y.Z "Release name goes here." released!
```
Comment (kellyselden) [inline RELEASE.md]: You can just link your tweet and the bot will fill in the text. Then people can easily retweet from slack.

**Comment by kellyselden** [comment]
added some notes

File: RELEASE.md [hunk 4]
```diff
@@ -76,6 +76,8 @@ Tag the release
 * click pre-release for beta releases
 * publish the release
 
+[Update ember-new-output and ember-addon-output.](https://gist.github.com/nathanhammond/e0a55b4d0328b45b8ef5)
```
Comment (kellyselden) [inline RELEASE.md]: Should probably also call out the additions to the changelog:
[`ember new` diff](https://github.com/ember-cli/ember-new-output/compare/v2.6.0-beta.2...v2.6.0-beta.3)
[`ember addon` diff](https://github.com/ember-cli/ember-addon-output/compare/v2.6.0-beta.2...v2.6.0-beta.3)

**Comment by nathanhammond** [comment]
Notes addressed. @kellyselden r+?

**Comment by kellyselden** [comment]
looks great! @homu r+

**Pull request merged**